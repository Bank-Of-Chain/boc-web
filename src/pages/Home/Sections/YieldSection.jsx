import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import CheckIcon from "@material-ui/icons/Check";
import styles from "./yieldStyle";

const useStyles = makeStyles(styles);

export default function YieldSection() {
  const classes = useStyles();
  const yieldSources = [
    "Market-making fees.",
    "Interest from over-collateralized lending.",
    "Governance token rewards.",
  ];
  return (
    <div className={classes.yieldSection}>
      <h2 className={classes.title}>
        Sources of
        <span className={classes.colorful}> yield.</span>
      </h2>
      <h3 className={classes.subTitle}>
        The best long-term <strong>risk-free</strong> return
      </h3>
      <p className={classes.description}>
        Bank of Chain maximizes returns in a scalable and safe manner, only
        interacting with carefully-vetted protocols.
      </p>
      <p className={classes.description}>
        Yields are obtained from three sources:
      </p>
      <ul className={classes.sourceList}>
        {yieldSources.map((item) => (
          <li key={item} className={classes.sourceItem}>
            <CheckIcon className={classes.checkIcon} />
            <span>{item}</span>
          </li>
        ))}
      </ul>
    </div>
  );
}
