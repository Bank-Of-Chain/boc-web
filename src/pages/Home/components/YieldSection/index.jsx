import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import CheckIcon from '@material-ui/icons/Check'
import styles from './yieldStyle'

const useStyles = makeStyles(styles)

export default function YieldSection() {
  const classes = useStyles()
  const yieldSources = ['Market-making fees.', 'Interest from over-collateralized lending.', 'Governance token rewards.', 'Rewards from staking.']
  return (
    <div className={classes.yieldSection}>
      <h2 className={classes.title}>
        Sources of
        <span className={classes.colorful}> yield</span>
      </h2>
      <h3 className={classes.subTitle}>
        The best long-term <strong>&quot;risk-free&quot;</strong> return
      </h3>
      <p className={classes.description}>
        Bank of Chain maximizes returns in a scalable and safe manner by only interacting with rigorously-evaluated protocols.
      </p>
      <p className={classes.description}>Yields are obtained from four sources:</p>
      <ul className={classes.sourceList}>
        {yieldSources.map(item => (
          <li key={item} className={classes.sourceItem}>
            <CheckIcon className={classes.checkIcon} />
            <span className={classes.text}>{item}</span>
          </li>
        ))}
      </ul>
    </div>
  )
}
