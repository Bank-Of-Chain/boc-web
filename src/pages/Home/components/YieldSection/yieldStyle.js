import { gradientText } from '@/assets/jss/common.js'

const yieldStyle = {
  yieldSection: {
    width: '80%',
    maxWidth: '1127px',
    margin: '0 auto',
    padding: '172px 0 198px'
  },
  title: {
    fontSize: '100px',
    fontWeight: 'bold',
    margin: 0
  },
  '@media screen and (max-width: 923px)': {
    title: {
      fontSize: '3.4rem'
    }
  },
  svg: {
    verticalAlign: 'middle'
  },
  colorful: {
    ...gradientText,
    fontWeight: 'bold'
  },
  text: {
    display: 'inline-block',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis'
  },
  subTitle: {
    fontSize: '26px',
    lineHeight: '36px',
    margin: '16px 0 32px'
  },
  description: {
    fontSize: '20px',
    margin: 0,
    lineHeight: '30px'
  },
  sourceList: {
    listStyle: 'none',
    padding: 0,
    marginTop: '36px'
  },
  sourceItem: {
    display: 'flex',
    alignItems: 'center',
    height: '72px',
    lineHeight: '72px',
    fontSize: '20px',
    borderTop: '1px solid #fff',
    '&:last-child': {
      borderBottom: '1px solid #fff'
    }
  },
  checkIcon: {
    fontSize: '28px',
    color: '#A68EFE',
    marginLeft: '2rem',
    marginRight: '3rem'
  }
}

export default yieldStyle
