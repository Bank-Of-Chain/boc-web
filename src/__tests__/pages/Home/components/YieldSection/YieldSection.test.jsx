import React from 'react'
import { render } from '@testing-library/react'
import YieldSection from '@/pages/Home/components/YieldSection'

test('YieldSection Component Render', () => {
  const { asFragment } = render(<YieldSection />)
  expect(asFragment()).toMatchSnapshot()
})
