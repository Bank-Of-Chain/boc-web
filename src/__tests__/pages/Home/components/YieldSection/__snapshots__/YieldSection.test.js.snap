// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YieldSection Component Render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-yieldSection-1"
  >
    <h2
      class="makeStyles-title-2"
    >
      Sources of
      <span
        class="makeStyles-colorful-4"
      >
         yield.
      </span>
    </h2>
    <h3
      class="makeStyles-subTitle-5"
    >
      The best long-term 
      <strong>
        risk-free
      </strong>
       return
    </h3>
    <p
      class="makeStyles-description-6"
    >
      Bank of Chain maximizes returns in a scalable and safe manner, only interacting with carefully-vetted protocols.
    </p>
    <p
      class="makeStyles-description-6"
    >
      Yields are obtained from three sources:
    </p>
    <ul
      class="makeStyles-sourceList-7"
    >
      <li
        class="makeStyles-sourceItem-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-checkIcon-9"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
        <span>
          Market-making fees.
        </span>
      </li>
      <li
        class="makeStyles-sourceItem-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-checkIcon-9"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
        <span>
          Interest from over-collateralized lending.
        </span>
      </li>
      <li
        class="makeStyles-sourceItem-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-checkIcon-9"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
        <span>
          Governance token rewards.
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
